ui/apps/everest/src/components/app-bar/user-icon/UserIcon.tsx

```diff
@@ -23,6 +33,15 @@ const AppBarUserIcon = () => {
   const handleClose = () => {
     setAnchorEl(null);
   };
+  const token = localStorage.getItem('everestToken');
+  const decoded = jwtDecode(token!) as UserToken;
+
+  const preferredUsername = decoded.preferred_username;
+  const name = decoded.name;
+  const email = decoded.email;
+  const sub = decoded.sub?.substring(0, decoded.sub.indexOf(':')) || '';
```

Comment:
If the sub doesn't contain a suffix that starts with `:` we should still show the entire sub.
e.g. the sub claim from external IdPs are usually just an a random ID (no `:login` suffix) and in that case we want to show the entire sub.

Reply:
I this should do the trick
```suggestion
  const sub = decoded.sub?.substring(0, decoded.sub.indexOf(':')) || decoded.sub;
```